File: java/IEC62325/MarketManagement/Auction.java
package IEC62325.MarketManagement;

import IEC61970.Base.Domain.String;
import IEC61970.Base.Core.IdentifiedObject;

/**
 * A class providing the identification and type of an auction.
 * @created 03-Jan-2024 1:58:51 PM
 */
public class Auction extends IdentifiedObject {

	/**
	 * Identification of the method of allocation in an auction.
	 */
	public String allocationMode;
	/**
	 * An indicator that signifies that the auction has been cancelled.
	 */
	public String cancelled;
	/**
	 * The product category of an auction.
	 */
	public String category;
	/**
	 * The terms which dictate the determination of the bid payment price.
	 */
	public String paymentTerms;
	/**
	 * The rights of use the transmission capacity acquired in an auction.
	 */
	public String rights;
	/**
	 * The kind of the Auction (e.g. implicit, explicit ...).
	 */
	public String type;
	public TimeSeries TimeSeries;

	public Auction(){

	}
}//end Auction

File: py/IEC62325/MarketManagement/Auction.py
from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC62325.MarketManagement.TimeSeries import TimeSeries

class Auction(IdentifiedObject):
    """
    A class providing the identification and type of an auction.
    """
    def __init__(self):
        super().__init__()  # Call to the superclass's constructor
        self.allocation_mode = str()  # Identification of the method of allocation in an auction.
        self.cancelled = str()  # An indicator that signifies that the auction has been cancelled.
        self.category = str()  # The product category of an auction.
        self.payment_terms = str()  # The terms which dictate the determination of the bid payment price.
        self.rights = str()  # The rights of use the transmission capacity acquired in an auction.
        self.type = str()  # The kind of the Auction (e.g. implicit, explicit ...).
        self.time_series = TimeSeries()  # Associated TimeSeries object
